frontend/src/components/filters/Filters.js

```diff
@@ -2,604 +2,133 @@ import counterpart from 'counterpart';
-    }
+  clearFilters = (filterToClear) => {
+    const { filterId, clearAllFilters, filters, updateDocList } = this.props;
+    clearAllFilters({ id: filterId, data: filterToClear });
+    // fetch again the doc content after filters were updated into the store
+    updateDocList(filters.filtersActive);
   };
 
   /**
    * @method dropdownToggled
-   * @summary ToDo: Describe the method
+   * @summary Resets notValidFields flag to false
    */
   dropdownToggled = () => {
-    this.setState({
-      notValidFields: false,
-    });
-  };
-
-  /**
-   * @method annotateFilters
-   * @summary I think it creates caption for active filters to show when the widget is closed - Kuba
-   * @param {array} unannotatedFilters
-   */
-  annotateFilters = (unannotatedFilters) => {
-    const { activeFilter } = this.state;
-
-    return unannotatedFilters.map((unannotatedFilter) => {
-      const parameter =
-        unannotatedFilter.parameters && unannotatedFilter.parameters[0];
-      const isActive = this.isFilterActive(unannotatedFilter.filterId);
-      const currentFilter = activeFilter
-        ? activeFilter.find((f) => f.filterId === unannotatedFilter.filterId)
-        : null;
-      const activeParameter =
-        parameter && isActive && currentFilter && currentFilter.parameters[0];
-
-      const filterType =
-        unannotatedFilter.parameters && activeParameter
-          ? unannotatedFilter.parameters.find(
-              (filter) => filter.parameterName === activeParameter.parameterName
-            )
-          : parameter && parameter.widgetType;
-
-      const captionValue = activeParameter
-        ? fieldValueToString({
-            fieldValue: activeParameter.valueTo
-              ? [activeParameter.value, activeParameter.valueTo]
-              : activeParameter.value,
-            fieldType: filterType,
-          })
-        : '';
-
-      return {
-        ...unannotatedFilter,
-        captionValue,
-        isActive,
-      };
-    });
+    const { updateNotValidFields, filterId } = this.props;
+    updateNotValidFields({ filterId, data: false });
   };
 
-  // RENDERING FILTERS -------------------------------------------------------
   /**
    * @method render
-   * @summary ToDo: Describe the method
+   * @summary Main render function - renders the filters
    */
   render() {
     const {
-      filterData,
       windowType,
       viewId,
       resetInitialValues,
       allowOutsideClick,
       modalVisible,
+      filters,
+      filterId,
     } = this.props;
 
-    const {
-      notValidFields,
-      widgetShown,
-      activeFilter,
-      activeFiltersCaptions,
-    } = this.state;
+    const widgetShown = filters ? filters.widgetShown : false;
+    const notValidFields = filters ? filters.notValidFields : false;
+
+    if (!filters || !viewId || !filters.filterData) return false;
+    const { filtersActive, filtersCaptions: activeFiltersCaptions } = filters;
+
+    const allFilters = annotateFilters({
```

Comment:
Why can't this be handled by redux but instead we'll be calling this on each render and create a new js object ? Seems like we have everything we need in the `filters` prop.

Reply:
![Screenshot 2020-10-07 at 10 03 08](https://user-images.githubusercontent.com/1708561/95298171-546bfa80-0884-11eb-8b7d-c38acced2e1a.png)
I have them mapped in the mapStateToProps but once I do that I get a DocList unit test failing :(

Reply:
Thanks for the help in fixing this @siemiatj